src/user.rs

```diff
@@ -116,30 +120,35 @@ impl<'a> SingleUserRequestBuilder<'a> {
         SingleUserRequestBuilder { client, id }
     }
 
-    /// Retrieve all tracks uploaded by the artist
+    /// Retrieve all tracks uploaded by the user
     ///
     /// Returns:
     ///     a list of tracks or an error if one occurred.
-    pub async fn tracks(&mut self) -> Result<Vec<Track>> {
+    pub async fn tracks(&mut self) -> Result<Option<Vec<Track>>> {
```

Comment:
I've changed the api to only return a `Vec` as the `Option` doesn't provide any benefit to the user.